lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -185,16 +207,25 @@ def calculate_percentage_change_in_last_iteration(allvecs):
         truth_data = truth_data.astype(np.float32)
         sqrt_pi = np.sqrt(np.pi).astype(np.float32)
 
+        forecast_predictor_data = (
+            np.around(forecast_predictor_data, decimals=self.decimals))
+        forecast_var_data = np.around(
+            forecast_var_data, decimals=self.decimals)
+        truth_data = np.around(truth_data, decimals=self.decimals)
+        sqrt_pi = np.around(sqrt_pi, decimals=self.decimals)
+        initial_guess = np.around(initial_guess, decimals=self.decimals)
```

Comment:
Rather than doing this, wouldn't it be more appropriate to just use the `tol` argument to `minimize`?
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html

Reply:
As mentioned elsewhere, I've set the inputs to be float64, rather than setting the inputs to float32 and then rounding. This seems like it is giving more stable solutions.